lib/components/narrative/printable/printable-itinerary.js

```diff
@@ -92,19 +134,49 @@ class AccessLeg extends Component {
   }
 
   render () {
-    const { leg } = this.props
+    const { configCompanies, customIcons, leg } = this.props
+
+    // calculate leg mode label in a special way for this component
+    let legModeLabel = getLegModeLabel(leg)
+
+    if (leg.rentedBike) {
+      // FIXME: Special case for TriMet that needs to be refactored to
+      // incorporate actual company.
+      legModeLabel = 'Ride BIKETOWN bike'
+    } else if (leg.rentedCar) {
+      // Add extra information to printview that would otherwise clutter up
+      // other places that use the getLegModeLabel function
+      const companiesLabel = getCompaniesLabelFromNetworks(
+        leg.from.networks,
+        configCompanies
+      )
+      legModeLabel = `Drive ${companiesLabel} ${leg.from.name}`
+    } else if (leg.rentedVehicle) {
+      const companiesLabel = getCompaniesLabelFromNetworks(
+        leg.from.networks,
+        configCompanies
+      )
+      legModeLabel = `Ride ${companiesLabel} eScooter`
+    }
+
```

Comment:
I'm worried that this block of code to determine the print label will be at risk of diverging from the block in PlaceRow. Even though they aren't returning the exact same label, there's some shared logic that I have a feeling we would want to co-locate:

https://github.com/opentripplanner/otp-react-redux/blob/bd9e51c915b5ec73895ee85506ceaf5cd4812c19/lib/components/narrative/line-itin/place-row.js#L170-L206

Case in point: I think the only reason this pull requests exists is because there is a significant split between the print and narrative itinerary components.

Reply:
I noticed that, but it has already been diverging due to different styling and the text requested from Myleen is diverging, so I opted to make things different. To begin with, the line-itin way of displaying things shows the itinerary as a graph with transfer points looking like nodes and having the header be the title of the node. The text in the RentedVehicleLeg does somewhat closely match, but should it be exactly the same? The print way of displaying things is more of a list style with the title of each thing being the overall instructions.

Reply:
I'm not suggesting that the text should match between the two, but rather that the logic that processes the leg and delivers the text for the component should be co-located in a single method. It's almost more of a code management strategy to improve the chances that when modifications are made to the line-itin component we also remember to make the some modifications to the print view.